### PR TITLE
Bail early in `WP_Image_Editor_Imagick->make_subsize()` if image is already the requested size

### PR DESCRIPTION
**Pull request by danielbachhuber**
See https://core.trac.wordpress.org/ticket/57370

### CODE REVIEW

**Comment by NekoJonez** [comment]
Is that new error translate-able?

**Comment by danielbachhuber** [comment]
> Is that new error translate-able?

@NekoJonez It's wrapped in `__()` so it should be? Is there something specific you're uncertain about?

**Comment by NekoJonez** [comment]
> > Is that new error translate-able?
> 
> @NekoJonez It's wrapped in `__()` so it should be? Is there something specific you're uncertain about?

Maybe I am confusing with textdomains and such.

**Review by adamsilverstein — APPROVED** [review]
LGTM!

**Comment by danielbachhuber** [comment]
Shipped in https://core.trac.wordpress.org/changeset/55278

File: src/wp-includes/class-wp-image-editor-imagick.php [hunk 1]
```diff
@@ -503,6 +503,10 @@ public function make_subsize( $size_data ) {
 			$size_data['crop'] = false;
 		}
 
+		if ( ( $this->size['width'] == $size_data['width'] ) && ( $this->size['height'] == $size_data['height'] ) ) {
```
Comment (bueltge) [inline src/wp-includes/class-wp-image-editor-imagick.php:506]: Shouldn't we use the strict way via `===` to matching strict the type as well?

Comment (danielbachhuber) [reply src/wp-includes/class-wp-image-editor-imagick.php:506]: @bueltge Good question.

I think I adapted the check from here:

https://github.com/WordPress/wordpress-develop/blob/e993bc5f739af0ac589049c1f102788f57a1983d/src/wp-includes/class-wp-image-editor-gd.php#L172-L174

We could:

```suggestion
		if ( ( $this->size['width'] === $size_data['width'] ) && ( $this->size['height'] === $size_data['height'] ) ) {
```

Or we could:

```suggestion
		if ( ( (int) $this->size['width'] === (int) $size_data['width'] ) && ( (int) $this->size['height'] === (int) $size_data['height'] ) ) {
```

I'm a little bit hesitant about any obscure bugs that might crop up, though.

What do you think? Is there some prior art we could follow?

Comment (bueltge) [reply src/wp-includes/class-wp-image-editor-imagick.php:506]: In my opinion, I would always use a strict result. So your proposal with strict equal `===` and defined type `int` is really helpful and the best way for me.

Comment (mukeshpanchal27) [reply src/wp-includes/class-wp-image-editor-imagick.php:506]: 2nd option look good to me.

Comment (desrosj) [reply src/wp-includes/class-wp-image-editor-imagick.php:506]: I think I prefer the first option. The function's docblock notes that width and height will be of type `int`, and when the `$size` property is set, the values are cast as `(int)`.

https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/class-wp-image-editor.php#L209-L214

It also looks like `$size['width']` and `$size['height']` do not get type casted anywhere in the Imagick or GD image editor classes. So option 2 would be inconsistent with the rest class. Using option 2 could potentially hide instances where a plugin incorrectly sets the value of size to be non-integer values. Granted it's likely there would be other problems that occur as a result, but I'd rather not risk hiding an instance where something is amiss.

Comment (danielbachhuber) [reply src/wp-includes/class-wp-image-editor-imagick.php:506]: Thanks, folks! Given @desrosj's points, I'm landing the first option with https://github.com/WordPress/wordpress-develop/pull/4036

**Pull request closed without merging**